src/main/java/org/skife/jdbi/v2/sqlobject/BindBean.java

```diff
@@ -25,4 +25,13 @@
 {
     String BARE_BINDING = "___jdbi_bare___";
     String value() default BARE_BINDING;
+
+    // This override allows for explicitly specifying the type used to invoke the bound bean's methods.  By default the type is
+    // inferred from the argument that the annotated method is called with.  Providing an override may be required in certain
+    // situations, where the annotated parameter type is accessible from the org.skife.jdbi.v2.sqlobject package but a subtype
+    // is not.  The result of attempting to bind an inaccessible subtype being used without providing this override is a
+    // java.lang.IllegalAccessException.
+    Class<?> typeOverride() default Default.class;
```

Comment:
Possibly `beanType` as a name instead of `typeOverride`?  I could go either way and am open to suggestions that would make the purpose of this parameter more explicit.